modules/caddytls/connpolicy.go

```diff
@@ -298,6 +311,13 @@ type ClientAuthentication struct {
 	// which are not in this list will be rejected.
 	TrustedLeafCerts []string `json:"trusted_leaf_certs,omitempty"`
```

Comment:
This is basically for setting up a LeafCertValidator right? I wonder if we should deprecate this field or something. (Pending on #4518)

Reply:
Yes, good point currently the LeafCertValidator is added programatically if TrustedLeafCerts are not empty (to be backward compatible)
But it would be an idea to extract the LeafCertValidator to an own Module so this configuration would then not be inside ClientAuthentication anymore but inside the Validators own config. Could also be extracted outside caddy.

Reply:
I might make a commit to this branch to mark this as deprecated, (or not) -- but I'll play with it a bit first.